internal/service/remotecfg/config_manager.go

```diff
@@ -391,6 +437,75 @@ func (cm *configManager) getRemoteHash() string {
 	return cm.remoteHash
 }
 
+// setRemoteConfigStatus updates the remote config status.
+func (cm *configManager) setRemoteConfigStatus(status collectorv1.RemoteConfigStatuses, errorMessage string) {
+	cm.mut.Lock()
+	defer cm.mut.Unlock()
+
+	cm.remoteConfigStatus = &collectorv1.RemoteConfigStatus{
+		Status:       status,
+		ErrorMessage: errorMessage,
+	}
+}
+
+// getRemoteConfigStatus returns a copy of the current remote config status.
+func (cm *configManager) getRemoteConfigStatus() *collectorv1.RemoteConfigStatus {
+	cm.mut.RLock()
+	defer cm.mut.RUnlock()
+
+	// Return a copy to avoid race conditions
+	return cm.copyRemoteConfigStatus()
+}
+
+// getRemoteConfigStatusForRequest returns the remote config status only if it should be sent
+// (first call or status has changed since last sent).
+func (cm *configManager) getRemoteConfigStatusForRequest() *collectorv1.RemoteConfigStatus {
+	cm.mut.Lock()
+	defer cm.mut.Unlock()
+
+	// Send status if we've never sent one before (first call) or if it has changed
```

Comment:
This comment doesn't accurately describe the other state, where we reverted to nil because of a failed request.